### ISSUE TITLE
Default Planks smooth shading causes curved boards.

### ISSUE THREAD

**Issue by woutervddn** [varkenvarken/blenderaddons#6]
By default planks uses the smooth shading setting to (presumably) overcome a weird look on the bevels when rendering. While this part goes well, it leaves us with boards that look slightly convex (rather than the concave option that is also present in the settings).

![Default 'smooth' floorboard](https://user-images.githubusercontent.com/1514131/29515598-572bace4-866d-11e7-8a88-0476079583de.png)

We can select the boards an go to Tool > Shading: Flat to overcome this, but this leaves us with not-so-great-looking beveled edges. Manually increasing the segments and changing the limit method to Angle does solve it somewhat. However, this not a complete solution for this issue.

![Flat floorboards](https://user-images.githubusercontent.com/1514131/29516200-cd8cd488-866f-11e7-850f-d8496be8b55d.png)

We ourselves usually use smooth together with supporting edge loops. One way to achieve this is by using the 'Edge Split' Modifier. This however breaks the bevel option since it's applied to all faces. Manually adding it to a huge amount of planks is not an option either.

So at this stage we're stuck between smoothed edges with concave planks or flat plank with hard edges.

I'm not totally sure what the best option for this might be. Judging by the commented out line you, @varkenvarken, aren't completely sure of the answer either. I think this thread could be a great place for discussing which option we can use to make this better (programmatically).

**Comment by varkenvarken** [comment]
Might adding a crease weight to selected edges solve the issue (because I guess the subdiv modifier is the actual culprit)?

**Comment by woutervddn** [comment]
We didn't use the hollow function. So there is no subdiv. mod. enabled on the planks.

**Comment by varkenvarken** [comment]
I see, I think I could add supporting edge loops programmatically but maybe
there are better options. I have to revisit my code because I am not
certain if the bevel operator or bevel modifier is used here. I will
investigate.

**Comment by woutervddn** [comment]
It used to use edge split, but it's using bevel instead now (according to the commented line in the code here: https://github.com/varkenvarken/blenderaddons/blob/2881ae234bf465227bc0e74f4a42bcb3846e6d49/planks.py#L770

One thing I noticed: If you're not using the hollow function, then the subdivision multiplier isn't used. Which leaves you with enough 'room' to increase the amount of segments on the bevel modifier. Going from 2 to 4 (or even 3) there seems to be good enough not to notice the bow in the long direction.

Maybe we can conditionally change that value? 
(I'll do some further testing together with our desigers)

**Comment by varkenvarken** [comment]
Hi Wouter,

one way of achieving smooth-but-flat planks is to add an inset to all top faces:

![screenshot_2017-08-27_08-24-47](https://user-images.githubusercontent.com/3612601/29747659-9f742178-8b01-11e7-8856-94caf184ef70.png)

The inset is flat (i.e. at the exact same height as the original top face) but the width is kept very small.

![image](https://user-images.githubusercontent.com/3612601/29747665-e284f122-8b01-11e7-90d4-cae35788a761.png)

So basically we add supporting edge loops around the edge of each face here. Is that what you are looking for? 
I am a bit confused here because a bevel  modifier can have almost the same effect but you mention you don't want it to be applied to all edges? But that is some that can be controlled by the 'sharp' attribute of an edge so that is in principle something we could manage too. So to which edges would you apply a bevel?
